src/cli/self_update.rs

```diff
@@ -382,9 +382,9 @@ macro_rules! post_install_msg_unix_source_env {
 the corresponding `env` file under {cargo_home}.
 
 This is usually done by running one of the following (note the leading DOT):
-    . "{cargo_home}/env"            # For sh/bash/zsh/ash/dash/pdksh
-    source "{cargo_home}/env.fish"  # For fish
-    source "{cargo_home}/env.nu"    # For nushell
+    . "{cargo_home}/env"                       # For sh/bash/zsh/ash/dash/pdksh
+    source "{cargo_home}/env.fish"             # For fish
+    source $"($nu.home-path)/.cargo/env.nu"    # For nushell
```

Comment:
Note that this is not necessary the `$CARGO_HOME`! It's just its default value...

Reply:
AIUI if `$CARGO_HOME` is something else then this will get out of sync with other commands.

Reply:
Is there a scenario where $CARGO_HOME is set to a value different than default on a system and a human user will actually read this prompt? Or is it something that's only overriden in CI and such?

Reply:
@Yethal A human user will actually read the prompt. We allow `$CARGO_HOME` to be set to something else beforehand, actually IIRC we have suggested in an earlier note (before the actual installation) that if you want it to be elsewhere you can set that and return to the installer.

Reply:
So willit suffice to add a note that if `$CARGO_HOME` was overriden then the path will need to be adjusted accordingly?

Reply:
@Yethal I am still inclined to ding this properly.

For example, you probably can get inspired from this logic:

https://github.com/rust-lang/rustup/blob/21d4b2777f5d77f9621912208cea29062fbefaf6/src/cli/self_update.rs#L469-L487

... and figure out a proper way to print the exact path for `env.nu` for nushell. Maybe instead of using `{cargo_home}`, you can add an extra format argument `{cargo_home_nushell}`.

The only thing I don't know is how `$nu.home-path` is defined in nushell. However the spirit stays the same: if dirname is `($nu.home-path)/.cargo` you can print `source $"($nu.home-path)/.cargo/env.nu" `, otherwise just print out the full path.

Reply:
Ok I think I have got the full picture. Nushell uses https://docs.rs/dirs/latest/dirs/fn.home_dir.html meanwhile we use https://docs.rs/home/latest/home/fn.home_dir.html for the `$HOME` definition, so it's a perfect match on Unix.

However, I'm not so sure about Windows as the former uses `{FOLDERID_Profile}` where as the latter uses `%USERPROFILE%` and tries to fall back on using `{FOLDERID_Profile}`. Would that cause inaccurate results? @ChrisDenton

Reply:
Afaict Rustup doesn't support shell customization on Windows anyway.

Reply:
@Yethal I see what you mean, however there is a slight chance that the user has manually set `%USERPROFILE` to something else. Maybe a special branch needs to be added in that case.

Reply:
If user has manually set `%USERPROFILE%` to something other than their profile then I'd say it's on them if the prompt is incorrect

Reply:
Might this be considered a nushell bug? I think they could use `std::env::home_dir` since the implementation was fixed in 1.85 (though it won't be undeprecated until 1.87).

Reply:
What, the fact that environment variables are not parse-time safe? To the best of my knowledge this is by design.

Reply:
> What, the fact that environment variables are not parse-time safe? To the best of my knowledge this is by design.

No, I believe @ChrisDenton meant that [dirs/fn.home_dir.html](https://docs.rs/dirs/latest/dirs/fn.home_dir.html)'s behavior could be considered a bug in nushell's case...

Reply:
No, I mean the fact that it only uses `{FOLDERID_Profile}` and doesn't try `%USERPROFILE%` first. That would at least mean the behaviour is consistent.

Reply:
If the behavior was fied in 1.83 then it might be too early for Nushell since the policy is to stay two versions behind latest stable